cmd/endpoint.go

```diff
@@ -200,6 +217,70 @@ func (endpoints EndpointList) GetString(i int) string {
 	return endpoints[i].String()
 }
 
+// UpdateIsLocal - resolves the host and discovers the local host.
+func (endpoints EndpointList) UpdateIsLocal() error {
+	bArr := make([]bool, len(endpoints))
+	var foundLocal bool
+	// Mark the starting time
+	startTime := time.Now()
+	for {
+		// Retry infinitely on Kubernetes and Docker swarm.
+		// This is needed as the remote hosts are sometime
+		// not available immediately.
+
+		for i, b := range bArr {
+			if b {
+				continue
+			}
+
+			if endpoints[i].IsLocal {
+				bArr[i] = true
+				foundLocal = true
+				continue
+			}
+
+			// return err if not Docker or Kubernetes
+			// We use IsDocker() method to check for Docker Swarm environment
+			// as there is no reliable way to clearly identify Swarm from
+			// Docker environment.
+			isLocal, err := isLocalHost(endpoints[i].normalizedHost)
+			if err == nil {
+				bArr[i] = true
+				endpoints[i].IsLocal = isLocal
+				if isLocal {
+					foundLocal = true
+				}
+				continue
+			} else if !IsDocker() && !IsKubernetes() {
+				return err
+			}
+
+			// time elapsed
+			timeElapsed := time.Since(startTime)
+			// log error only if more than 1s elapsed
+			if timeElapsed > time.Second {
+				// log the message to console about the host not being
+				// resolveable.
+				reqInfo := (&logger.ReqInfo{}).AppendTags("host", endpoints[i].normalizedHost)
+				reqInfo.AppendTags("elapsedTime", humanize.RelTime(startTime, startTime.Add(timeElapsed), "elapsed", ""))
+				ctx := logger.SetReqInfo(context.Background(), reqInfo)
+				logger.LogIf(ctx, err)
+			}
+
+		}
+		// Break if the local endpoint is found already.
+		if foundLocal || (!IsDocker() && !IsKubernetes()) {
+			break
+		}
+		fmt.Println(endpoints)
```

Comment:
is this PR ready? @Praveenrajmani i see statements like this.

Reply:
yes it is ready. These log prints are by mistake. i will remove @harshavardhana

Reply:
And there are few code improvisations in the UpdateIsLocal to be done. I will update that tomorrow

Reply:
Tagging this PR as a WIP till then